dgl_ptm/dgl_ptm/model/initialize_model.py

```diff
@@ -100,42 +111,63 @@ class PovertyTrapModel(Model):
                             'deletion_prob':0.05,
                             'ratio':0.1,
                             'homophily_parameter':0.69,
-                            'characteristic_distance':35, 
+                            'characteristic_distance':35,
                             'truncation_weight':1.0e-10,}
     }
 
-    def __init__(self,*, model_identifier=None, restart=False, savestate=None):
+    def __init__(self,*, model_identifier, restart=None, savestate=True):
```

Comment:
We probably want to store with the saved state, the step that the process was in when the state was saved.
Then, we can have the process re-start from either a specific step (restart=X, where X is that specific step), or restart from the latest stored step (restart=-1), or start a new process (restart=None; fail on existing model storage path).

Reply:
Additionally, we still probably want to store in some way the current version of the code used to create the save state, as opposed to only the version of the model.

We may be able to do this by adding a (text) file to the program that contains a git hash that can be compared to a hash stored in the model log. This git hash file can then for example be updated on every commit, or on release.

Not sure whether differing hashes should throw a warning or an error. Also, we should probably discuss this with the researchers, because they may only really be concerned with the model version.

Reply:
I like the idea of linking code versions via a git hash. In general researchers will probably be mostly concerned with model version, but if you want to go for full-fledged FAIR, then code version would also be necessary. IMO a warning should suffice on mismatch.

Reply:
We can also store the version of `dgl_ptm` next to the model files. I noticed that `data_collection` module also stores some data. I created issue #52 for adding software version either via version or git hash to all information stored by the model.